tcms/xmlrpc/api/testcase.py

```diff
@@ -879,20 +877,25 @@ def link_plan(request, case_ids, plan_ids):
             err_msg = 'TestPlan %s does not exist.' % ids_str
         raise ObjectDoesNotExist(err_msg)
 
+    # (plan_id, case_id) pair might probably exist in test_case_plans table, so
+    # skip the ones that do exist and create the rest.
+    # note: this query returns a list of tuples!
+    existing = TestCasePlan.objects.filter(
+        plan__in=plan_ids,
+        case__in=case_ids
+    ).values_list('plan', 'case')
+
     # Link the plans to cases
     def _generate_link_plan_value():
         for plan_id in plan_ids:
             for case_id in case_ids:
-                yield plan_id, case_id
+                if (plan_id, case_id) not in existing:
```

Comment:
> There is a unique constraint in table to avoid inserting duplicate pair of plan and case

Yes, this is what "INSERT IGNORE" was used for, right ?

Because the ORM doesn't have such equivalent I am replacing the "INSERT IGNORE" with a new ORM query which selects the existing IDs from the DB and this check which prevents the `bulk_create()` method to attempt and insert duplicate records. There is no change in existing behavior, only in existing implementation.  That's why there is the `test_insert_ignores_existing_mappings` test case. Does this answer your question ?

And yes I am using the XML-RPC extensively.